### PR TITLE
Correctly build whats-new with typescript

### PR DESCRIPTION
**Pull request by noahtallen**
## Proposed Changes
While working on #78711, I was troubleshooting a type error in this package and noticed that while it's written entirely in TS, the build system doesn't reflect that. I made various changes to the package to match other packages which are build by TS. Additionally, I removed the typings file, since none of it is needed any more.

## Testing Instructions
1. `cd packages/whats-new && yarn build` -- does it work, and do files in `dist` look correct?
2. While I originally had testing instructions for testing the feature, this PR didn't cause an ETK release build to get generated! That means the built files are identical, which tells me this is working well.

## To do:
- [x] After #78711 is merged, try removing the whats-new type declaration which extends `HTMLElement`.

### CODE REVIEW

**Review by tyxla — COMMENTED** [review]
Looking good to me, but I agree - let's wait for #78711 to land first.

File: packages/whats-new/types.d.ts [hunk 1]
```diff
@@ -1,12 +0,0 @@
-// TODO: Attempt removing after https://github.com/Automattic/wp-calypso/pull/78711
-// Fixed node_modules/@wordpress/components/build-types/dropdown-menu/types.d.ts(43,24): error TS2339: Property 'role' does not exist on type 'HTMLElement'.
-interface HTMLElement {
-	role: string;
```
Comment (noahtallen) [inline packages/whats-new/types.d.ts:4]: Hm, looks like we still need this. I'm pretty confused why adding it in whats-new fixes it, so I want to dig into the issue further.

**Comment by noahtallen** [comment]
Turns out that we just needed to update Typescript and it works now :) I think that makes sense, as TS tweaked/updated a lot of DOM types over the several releases we just updated.

**Comment by noahtallen** [comment]
I'm going to move forward with this, since CI has verified the major pieces! Happy to follow-up with feedback

**Pull request merged**